rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,982 @@
+        DESTINATION_ADDRESS and TRANSACTION_ID:
+          - Fiat movement: both attributes won't show any value.
+          - Cryptocurrency movement: address and tx id on the blockchain.
+
+        Timestamp is from MTS_STARTED (when the movement was created), and not
+        from MTS_UPDATED (when it was completed/cancelled).
+
+        Can raise DeserializationError.
+
+        Schema reference in:
+        https://docs.bitfinex.com/reference#rest-auth-movements
+        """
+        if raw_result[9] != 'COMPLETED':
+            raise DeserializationError(
+                f'Unexpected bitfinex movement with status: {raw_result[5]}. '
+                f'Only completed movements are processed. Raw movement: {raw_result}',
+            )
+        try:
+            fee_asset = asset_from_bitfinex(
+                bitfinex_name=raw_result[1],
+                currency_map=currency_map,
+            )
+        except (UnknownAsset, UnsupportedAsset) as e:
+            asset_tag = 'Unknown' if isinstance(e, UnknownAsset) else 'Unsupported'
+            raise DeserializationError(
+                f'{asset_tag} {e.asset_name} found while processing movement asset '
+                f'due to: {str(e)}',
+            ) from e
+
+        amount = deserialize_asset_amount(raw_result[12])
+        category = (
+            AssetMovementCategory.DEPOSIT
+            if amount > ZERO
+            else AssetMovementCategory.WITHDRAWAL
+        )
+        address = None
+        transaction_id = None
+        if fee_asset.is_fiat() is False:
+            address = raw_result[16]
+            transaction_id = raw_result[20]
+
+        asset_movement = AssetMovement(
+            timestamp=Timestamp(int(raw_result[5] / 1000)),
+            location=Location.BITFINEX,
+            category=category,
+            address=address,
+            transaction_id=transaction_id,
+            asset=fee_asset,
+            amount=abs(amount),
+            fee_asset=fee_asset,
+            fee=Fee(abs(deserialize_fee(raw_result[13]))),
+            link=str(raw_result[0]),
+        )
+        return asset_movement
+
+    @staticmethod
+    def _deserialize_trade(
+            raw_result: List[Any],
+            currency_map: Dict[str, str],
+    ) -> Trade:
+        """Process a trade result from Bitfinex and deserialize it.
+
+        The base and quote assets are instantiated using the `fee_currency_symbol`
+        (from raw_result[10]) over the pair (from raw_result[1]).
+
+        Known pairs format: 'tETHUST', 'tETH:UST'.
+
+        Can raise DeserializationError.
+
+        Schema reference in:
+        https://docs.bitfinex.com/reference#rest-auth-trades
+        """
+        amount = deserialize_asset_amount(raw_result[4])
+        trade_type = TradeType.BUY if amount >= ZERO else TradeType.SELL
+        bfx_pair = raw_result[1].replace(':', '')
+        if bfx_pair.startswith('t'):  # Don't replace 't' just in case it appeared within the pair
```

Comment:
super nitpick: No need for this comment. Ofcourse you don't do a blind string replacement!